### PR TITLE
fix: 🐛 treat isActive to case-insensitive

### LINKED ISSUE #3656: case-insensitive on isActive property
### Version
3.5.3

### Reproduction link
[localhost/active-links/abouT](http://localhost:8080/active-links/abouT)

### Steps to reproduce
- Run `yarn run dev` in the root path of the project
- Go to http://localhost:8080/active-links/abouT

### What is expected?
The "/about" will be treated as active.

### What is actually happening?
The "/about" is not treated as active.

---
Since Vue Router already supports case-insensitive matching for routes themselves. I think it should be the same for the active match to be case-insensitive too.

### PR DESCRIPTION
**Pull request by Kyle-Ye**
✅ Closes: #3656

### CODE REVIEW

**Comment by Kyle-Ye** [comment]
Maybe we need to add some tests for this new behavior, any suggestion where to write the new test?

**Review by posva — CHANGES_REQUESTED** [review]
This is not the correct patch as there is an option to make the path matching case sensitive and this should respect it.
This information should be available on the route records as `regex.ignoreCase` or something similar.
Can you also add a few unit tests for isIncludeRoute?

**Comment by Kyle-Ye** [comment]
> This is not the correct patch as there is an option to make the path matching case sensitive and this should respect it.
> This information should be available on the route records as `regex.ignoreCase` or something similar.
> Can you also add a few unit tests for isIncludeRoute?


Add the test and use regex.ignoreCase property. Any suggestion on how to safely get ignoreCase properly?

File: src/components/link.js [hunk 1]
```diff
@@ -72,10 +72,11 @@ export default {
       ? createRoute(null, normalizeLocation(route.redirectedFrom), null, router)
       : route
 
+    const ignoreCase = current.matched[0].regex.ignoreCase
```
Comment (Kyle-Ye) [inline src/components/link.js:75]: Any suggestion on how to safely get ignoreCase properly? @posva

**Comment by Kyle-Ye** [comment]
Hello @posva. Any suggestion on this PR?

**Pull request closed without merging**